### PR TITLE
Adapt to core changes (addon.xml)

### PR DESCRIPTION
**Pull request by J-N-K**
See https://github.com/openhab/openhab-core/pull/3050

This adapts to the changes in core and adds metadata for all add-ons

### CODE REVIEW

**Comment by ssalonen** [comment]
Are all bundles truly included? Should they be? I searched with Bluetooth and modbus and could only find the "main" addon but not the children.

**Comment by J-N-K** [comment]
There should be one `addon.xml` for each add-on, not for each bundle. 

I believe that bluetooth and modus also share one feature, are shown as one add-on in the UI and all sub-bundles are always installed together with the main bundle. So it is correct that they only have one `addon.xml` as they are treated as one add-on by openHAB.

The XMLs for `io`, `transform` and `voice` add-ons are still missing (also for `ui`, but they are in a different repo).

**Review by druciak — APPROVED** [review]
Approving changes for Satel binding

**Comment by andrewfg** [comment]
@J-N-K is there an `addon.xsd` schema file available for this?

(.. the link https://www.openhab.org/schemas/addon-1.0.0.xsd isn't ..)

**Comment by J-N-K** [comment]
@andrewfg See here: https://github.com/openhab/openhab-core/pull/3050/files#diff-f0767731a7096c72866698cc8bc9378d10c340256fb65775c111f2b80cbf89b1

**Comment by ssalonen** [comment]
@J-N-K thanks for checking, with your explanation it sounds correct 👍

**Comment by andrewfg** [comment]
> See here

Ok, so essentially `addon` is a superset of `binding` where all the added elements in `addon` have 'minOccurs:0' ..
... therefore LGTM for me for NeoHub / Siemens RDS / HDPowerView / Tado / Velux ..

**Review by andrewfg — APPROVED** [review]

**Review by pacive — APPROVED** [review]

**Review by lsiepel — APPROVED** [review]
LGTM for PlugwiseHA

**Review by coop-git — APPROVED** [review]

**Review by mlobstein — APPROVED** [review]

**Review by CSchlipp — APPROVED** [review]
👍🏼 for WlanThermo

**Review by psmedley — APPROVED** [review]

**Review by marcusb — APPROVED** [review]

**Review by ArjenKorevaar — APPROVED** [review]

**Review by appzer — APPROVED** [review]
Pushsafer approved

**Review by jeremystucki — APPROVED** [review]

**Review by hmerk — APPROVED** [review]
LGTM

**Review by marcelrv — APPROVED** [review]
LGTM for max, miio, windcentrale,ntp

**Review by andibraeu — APPROVED** [review]

**Review by mherwege — APPROVED** [review]

**Review by EvilPingu — APPROVED** [review]

**Review by renescherer — APPROVED** [review]

**Review by janvyb — APPROVED** [review]

**Review by mvalla — APPROVED** [review]

**Review by MatthiasKlocke — APPROVED** [review]

**Review by kjoglum — APPROVED** [review]

**Review by olivierkeke — APPROVED** [review]

File: bundles/org.openhab.binding.linuxinput/src/main/resources/OH-INF/addon/addon.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon:addon id="linuxinput" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:addon="https://openhab.org/schemas/addon/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/addon/v1.0.0 https://openhab.org/schemas/addon-1.0.0.xsd">
+
+	<type>binding</type>
+	<name>LinuxInput Binding</name>
+	<description>This is the binding for LinuxInput.</description>
```
Comment (t-8ch) [inline bundles/org.openhab.binding.linuxinput/src/main/resources/OH-INF/addon/addon.xml:8]: "Exposes input events like keypresses from the Linux input subsystem to OpenHAB"

**Review by t-8ch — APPROVED** [review]

**Review by dalgwen — APPROVED** [review]

File: bundles/org.openhab.automation.groovyscripting/src/main/resources/OH-INF/addon/addon.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon:addon id="groovyscripting" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:addon="https://openhab.org/schemas/addon/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/addon/v1.0.0 https://openhab.org/schemas/addon-1.0.0.xsd">
+
+	<type>automation</type>
+	<name>Groovy Scripting</name>
+	<description>This is adds a Groovy script engine.</description>
```
Comment (jlaur) [inline bundles/org.openhab.automation.groovyscripting/src/main/resources/OH-INF/addon/addon.xml:8]: Here and for most other automations in commit: "This is adds".

Comment (J-N-K) [reply bundles/org.openhab.automation.groovyscripting/src/main/resources/OH-INF/addon/addon.xml:8]: Fixed

**Review by kabili207 — APPROVED** [review]

**Review by bobadair — APPROVED** [review]
LGTM for Lutron and Alarmdecoder bindings.

**Review by sgiehl — APPROVED** [review]
LGTM for Luxtronik binding

**Review by duhast — APPROVED** [review]

**Review by yfre — APPROVED** [review]
LGTM for homekit

**Review by Cossey — APPROVED** [review]
Approved for:
HCC Rubbish Collection
HP Printer
NZ Water Alerts

**Review by crnjan — APPROVED** [review]

**Review by robnielsen — APPROVED** [review]

**Comment by holgerfriedrich** [comment]
@J-N-K is the addon id intended to be unique?
If yes, we seem to have a naming conflict for the following addons:
org.openhab.binding.exec / org.openhab.transform.exec
org.openhab.binding.neeo / org.openhab.io.neeo

```
<?xml version="1.0" encoding="UTF-8"?>
<addon:addon id="exec" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:addon="https://openhab.org/schemas/addon/v1.0.0"
	xsi:schemaLocation="https://openhab.org/schemas/addon/v1.0.0 https://openhab.org/schemas/addon-1.0.0.xsd">

	<type>binding</type>
	<name>Exec Binding</name>
	<description>This is the binding to execute arbitrary shell commands</description>

</addon:addon>
```

```
<?xml version="1.0" encoding="UTF-8"?>
<addon:addon id="exec" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:addon="https://openhab.org/schemas/addon/v1.0.0"
	xsi:schemaLocation="https://openhab.org/schemas/addon/v1.0.0 https://openhab.org/schemas/addon-1.0.0.xsd">

	<type>transformation</type>
	<name>Exec transformation</name>
	<description>Transforms an input string with an external program.</description>

</addon:addon>
````

**Comment by J-N-K** [comment]
If I didn't make any mistake, only the UID (i.e. `type-id` - `binding-knx`) should be used for identification. This is also unique across different add-on services because non-distribution add-on services use a prefix (like `marketplace:`).

**Review by stefan-hoehn — APPROVED** [review]

**Review by kaikreuzer — APPROVED** [review]

**Comment by kaikreuzer** [comment]
There still seems to be a reference to the old bundle in the itests:
```
[ERROR] Error   : org.openhab.core.binding.xml;version=[4.0.0,4.0.1) Not found in [bnd-cache [/home/jenkins/jenkins-agent1/workspace/PR-openHAB-Addons/itests/org.openhab.binding.feed.tests
```

**Review by druciak — APPROVED** [review]

**Comment by J-N-K** [comment]
My fault. I forgot to resolve after I rebased. Give me a minute.

File: bundles/org.openhab.automation.jsscripting/src/main/resources/OH-INF/addon/addon.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon:addon id="jsscripting" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:addon="https://openhab.org/schemas/addon/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/addon/v1.0.0 https://openhab.org/schemas/addon-1.0.0.xsd">
+
+	<type>automation</type>
+	<name>JavaScript Scripting</name>
+	<description>This adds a JS (ECMAScript-2021) script engine.</description>
```
Comment (florian-h05) [inline bundles/org.openhab.automation.jsscripting/src/main/resources/OH-INF/addon/addon.xml:8]:
```suggestion
	<description>This adds a JS (ECMAScript-2022) script engine.</description>
```

**Review by florian-h05 — APPROVED** [review]
JavaScript Scripting and evcc ✅
Just some minor suggestion for the description.

**Review by hmerk — APPROVED** [review]
LGTM for ekey, energenie, intesis and wemo binding.

**Review by kabili207 — APPROVED** [review]
LIRC looks good to me

**Comment by morph166955** [comment]
This seems to have potentially broken the neeo binding.  As of this afternoon when I reset my docker container I get:

2023-01-15 21:17:32.657 [ERROR] [core.karaf.internal.FeatureInstaller] - Failed to refresh bundles after processing config update
org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=openhab-misc-neeo; type=karaf.feature; version="[4.0.0.SNAPSHOT,4.0.0.SNAPSHOT]"; filter:="(&(osgi.identity=openhab-misc-neeo)(type=karaf.feature)(version>=4.0.0.SNAPSHOT)(version<=4.0.0.SNAPSHOT))" [caused by: Unable to resolve openhab-misc-neeo/4.0.0.SNAPSHOT: missing requirement [openhab-misc-neeo/4.0.0.SNAPSHOT] osgi.identity; osgi.identity=org.openhab.io.neeo; type=osgi.bundle; version="[4.0.0.202301152041,4.0.0.202301152041]"; resolution:=mandatory [caused by: Unable to resolve org.openhab.io.neeo/4.0.0.202301152041: missing requirement [org.openhab.io.neeo/4.0.0.202301152041] osgi.service; filter:="(objectClass=org.openhab.core.addon.AddonInfoRegistry)"; effective:=active]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[org.eclipse.osgi-3.17.200.jar:?]
        at org.apache.felix.resolver.ResolverImpl.doResolve(ResolverImpl.java:433) ~[org.eclipse.osgi-3.17.200.jar:?]
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:420) ~[org.eclipse.osgi-3.17.200.jar:?]
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:374) ~[org.eclipse.osgi-3.17.200.jar:?]
        at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:257) ~[?:?]
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:399) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1069) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:1004) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve openhab-misc-neeo/4.0.0.SNAPSHOT: missing requirement [openhab-misc-neeo/4.0.0.SNAPSHOT] osgi.identity; osgi.identity=org.openhab.io.neeo; type=osgi.bundle; version="[4.0.0.202301152041,4.0.0.202301152041]"; resolution:=mandatory [caused by: Unable to resolve org.openhab.io.neeo/4.0.0.202301152041: missing requirement [org.openhab.io.neeo/4.0.0.202301152041] osgi.service; filter:="(objectClass=org.openhab.core.addon.AddonInfoRegistry)"; effective:=active]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[org.eclipse.osgi-3.17.200.jar:?]
        ... 12 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve org.openhab.io.neeo/4.0.0.202301152041: missing requirement [org.openhab.io.neeo/4.0.0.202301152041] osgi.service; filter:="(objectClass=org.openhab.core.addon.AddonInfoRegistry)"; effective:=active
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[org.eclipse.osgi-3.17.200.jar:?]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341) ~[org.eclipse.osgi-3.17.200.jar:?]
        ... 12 more

**Comment by J-N-K** [comment]
Are you using latest core snapshot?

**Comment by morph166955** [comment]
I'm good now. Apparently I was one one snapshot behind where I needed to be. May not be a bad idea to post something on the forum so people know that their installs may randomly break due to this.

**Comment by jlaur** [comment]
@J-N-K - do you think you can create a script for fixing the I18N properties files?

Currently:

```
# binding

binding.xxx.name = XXX Binding
binding.xxx.description = This is the binding for XXX
```

Should be:

```
# addon

addon.xxx.name = XXX Binding
addon.xxx.description = This is the binding for XXX
```

**Comment by J-N-K** [comment]
I did the PR with the IDE: „replace in files“, filter „.properties“, replace „binding.“ with „addon.“, „replace all“.

**Comment by lolodomo** [comment]
@J-N-K @jlaur : take care, I believe the addon.xml file was forgotten for the nobohub binding.

**Comment by jlaur** [comment]
@lolodomo - looks like you are right:
image

**Comment by Sonic-Amiga** [comment]
Hi all! Sorry for being late, but okay...

**Pull request merged**